src/components/app/ProfileViewer.js

```diff
@@ -5,104 +5,51 @@
-import MarkerChart from '../marker-chart/';
-import FlameGraph from '../flame-graph/';
-import { changeSelectedTab, changeTabOrder } from '../../actions/app';
 import { returnToZipFileList } from '../../actions/zipped-profiles';
-import { getTabOrder } from '../../reducers/profile-view';
-import { getSelectedTab, getProfileName } from '../../reducers/url-state';
+import { getHiddenThreads, getProfileName } from '../../reducers/url-state';
+import { getThreads } from '../../reducers/profile-view';
 import ProfileViewerHeader from '../header/ProfileViewerHeader';
-import CallNodeContextMenu from '../shared/CallNodeContextMenu';
-import MarkerTableContextMenu from '../marker-table/ContextMenu';
-import ProfileThreadHeaderContextMenu from '../header/ProfileThreadHeaderContextMenu';
-import { toValidTabSlug } from '../../utils/flow';
 import { getHasZipFile } from '../../reducers/zipped-profiles';
 
-import type { Tab } from './TabBar';
 import type {
   ExplicitConnectOptions,
   ConnectedProps,
 } from '../../utils/connect';
-import type { TabSlug } from '../../types/actions';
 
 require('./ProfileViewer.css');
 
 type StateProps = {|
-  +tabOrder: number[],
-  +selectedTab: TabSlug,
   +profileName: string | null,
   +hasZipFile: boolean,
+  +visibleThreadsSize: number,
 |};
 
 type DispatchProps = {|
-  +changeSelectedTab: typeof changeSelectedTab,
-  +changeTabOrder: typeof changeTabOrder,
   +returnToZipFileList: typeof returnToZipFileList,
 |};
 
 type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
 
 class ProfileViewer extends PureComponent<Props> {
-  _tabs: Tab[];
-
-  constructor(props) {
-    super(props);
-    (this: any)._onSelectTab = this._onSelectTab.bind(this);
-    // If updating this list, make sure and update the tabOrder reducer with another index.
-    this._tabs = [
-      {
-        name: 'calltree',
-        title: 'Call Tree',
-      },
-      {
-        name: 'flame-graph',
-        title: 'Flame Graph',
-      },
-      {
-        name: 'stack-chart',
-        title: 'Stack Chart',
-      },
-      {
-        name: 'marker-chart',
-        title: 'Marker Chart',
-      },
-      {
-        name: 'marker-table',
-        title: 'Marker Table',
-      },
-      {
-        name: 'network-chart',
-        title: 'Network',
-      },
-    ];
-  }
-
-  _onSelectTab(selectedTab: string) {
-    const { changeSelectedTab } = this.props;
-    const tabSlug = toValidTabSlug(selectedTab);
-    if (!tabSlug) {
-      throw new Error('Attempted to change to a tab that does not exist.');
-    }
-    changeSelectedTab(tabSlug);
-  }
-
   render() {
     const {
-      tabOrder,
-      changeTabOrder,
-      selectedTab,
       hasZipFile,
       profileName,
       returnToZipFileList,
+      visibleThreadsSize,
     } = this.props;
+
+    // Each thread is 50px high, the time ruler is 20px.
+    // Eventually we might use a more automatic way, see
+    // https://github.com/zesik/react-splitter-layout/issues/20
```

Comment:
Could we file up a follow up issue for that when we merge? Thanks! :)

Reply:
Would you rather that we file a bug in perf.html now, or wait until that bug is actually fixed? I believe that it's good enough to have the reference to the bug in our source code, but please tell me what you think.

Reply:
The comments didn't sound like it will be fixed anytime soon, so maybe filing one could be good so we don't forget?

Reply:
My concern is we already have a lot of actionable issues that we don't look at, to add a non-actionable issue. I mean that I'm not sure adding this issue will actually help us remember it... more than the simple comment we already have here.

But I can add still it. (I mean, later, when we'll resume the work on this...)